homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -0,0 +1,382 @@
+        return self._mode
+
+    @property
+    def min_humidity(self):
+        """Return the minimum target humidity."""
+        return self._min_humidity
+
+    @property
+    def max_humidity(self):
+        """Return the maximum target humidity."""
+        return self._max_humidity
+
+    async def async_turn_on(
+        self,
+        **kwargs,
+    ) -> None:
+        """Turn the device on."""
+        result = await self._try_command(
+            "Turning the miio device on failed.", self._device.on
+        )
+        await self.coordinator.async_request_refresh()
+
+        if result:
+            self._state = True
+            self._skip_update = True
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn the device off."""
+        result = await self._try_command(
+            "Turning the miio device off failed.", self._device.off
+        )
+        await self.coordinator.async_request_refresh()
+
+        if result:
+            self._state = False
+            self._skip_update = True
+
+    def translate_humidity(self, humidity):
+        """Translate the target humidity to the first valid step."""
+        return (
+            math.ceil(percentage_to_ranged_value((1, self._humidity_steps), humidity))
+            * 100
+            / self._humidity_steps
+            if 0 < humidity <= 100
+            else None
+        )
+
+
+class XiaomiAirHumidifier(XiaomiGenericHumidifierDevice, HumidifierEntity):
+    """Representation of a Xiaomi Air Humidifier."""
+
+    def __init__(self, name, device, entry, unique_id, coordinator):
+        """Initialize the plug switch."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        if self._model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            self._available_modes = []
+            self._available_modes = [
+                mode.name
+                for mode in AirhumidifierOperationMode
+                if mode is not AirhumidifierOperationMode.Strong
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 10
+        elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
+            self._available_modes = [
+                mode.name for mode in AirhumidifierMiotOperationMode
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 100
+        else:
+            self._available_modes = [
+                mode.name
+                for mode in AirhumidifierOperationMode
+                if mode is not AirhumidifierOperationMode.Auto
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 10
+
+        self._state_attrs.update(
+            {attribute: None for attribute in self._available_attributes}
+        )
+        self.coordinator.async_add_listener(self.update)
+
+    @callback
+    def update(self):
```

Comment:
It's best to use a different name. `update` has special features in some entities.
```suggestion
    def _update(self):
```

Reply:
Done